src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,51 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  constructor(name, health = 100) {
+    this.name = name;
+    this.health = health;
+    this.addAnimal(this);
+    this.removeDeadAnimal();
+  }
+
+  addAnimal(animal) {
+    Animal.alive.push(animal);
+  }
+
+  removeDeadAnimal() {
+    Animal.alive = Animal.alive.filter(animal => animal.health >= 0);
+  }
+
+  decreaseHealth(amount) {
+    this.health -= amount;
+
+    if (this.health <= 0) {
+      const index = Animal.alive.indexOf(this);
+
+      Animal.alive.splice(index, 1);
+    }
+  }
 }
 
+Animal.alive = [];
+
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name, hidden = false) {
+    super(name);
+    this.hidden = hidden;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(animal) {
+    if (!(animal.hidden === true) && animal instanceof Herbivore) {
```

Comment:
```suggestion
    if (!animal.hidden && animal instanceof Herbivore) {
```

Same but a bit shorter